external-crates/move/move-compiler/src/parser/syntax.rs

```diff
@@ -2435,7 +2435,12 @@ fn parse_constant_decl(
     }
     consume_token(context.tokens, Tok::Const)?;
     let name = ConstantName(parse_identifier(context)?);
-    consume_token(context.tokens, Tok::Colon)?;
+    if consume_token(context.tokens, Tok::Colon).is_err() {
```

Comment:
Alternatively, we might want to make a helper `consume_token!(context.tokens, Tok::Colon, "Add type annotation to this constant")`
or `consume_token!(context.tokens, Tok::Colon, "Add type annotation to this constant")`

`consume_token_with(context.tokens, Tok::Colon, || "Expected a type annotation for this constant")`